File: tests/scripting/09_hash_sum.lua
-- Populate a hash and sum the values.
redis.call('DEL', KEYS[1]);
local n = tonumber(ARGV[1]) or 100;
for i = 1, n do
  redis.call('HSET', KEYS[1], 'f' .. i, i);
end
local sum = 0;
for i = 1, n do
  local v = redis.call('HGET', KEYS[1], 'f' .. i);
  sum = sum + tonumber(v);
end
return sum;

File: tests/scripting_js_faithful/09_hash_sum.js
// Faithful copy of Lua test 09_hash_sum.lua
// Expected: sum of 1..n
redis.call('DEL', KEYS[1]);
var n = Number(ARGV[1] || 100);
for (var i = 1; i <= n; i += 1) {
  redis.call('HSET', KEYS[1], 'f' + i, i);
}
var sum = 0;
for (var j = 1; j <= n; j += 1) {
  var v = redis.call('HGET', KEYS[1], 'f' + j);
  sum += Number(v);
}
return sum;
